hs_core/templates/pages/genericresource.html

```diff
@@ -1494,10 +1494,10 @@ <h4 class="modal-title">Confirm files deletion</h4>
                 }
 
                 // If does not exist, add it
-                if (!exists && keyword[i] != "") {
+                if (!exists && keyword[i] != "" && keyword[i].length <= 100) {
```

Comment:
All these checks are necessary to minimize failure of keyword update. The problem with keyword update is that we delete all existing keywords before the update. If the update fails for some reason there is the chance that we lose the existing keywords.